src/test/java/com/digirati/taxonomy/manager/normalisation/TextNormaliserTest.java

```diff
@@ -0,0 +1,111 @@
+package com.digirati.taxonomy.manager.normalisation;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class TextNormaliserTest {
+
+    @Test
+    void normaliseShouldLemmatizeText() {
```

Comment:
We shouldn't be testing specific lemmatization results, that can be changed on a whim by the underlying library (and we don't particularlycare if it does).

Reply:
These were about trying to document the behaviour of the library we're relying on, but I take your point that we don't really care about the specifics of it. I'll drop the tests about contractions and US/UK english, but since we're relying on the fact that some sort of lemmatisation occurs I think we should leave the happy-path test in place.

Reply:
If the libraries we are using change the lemmatization results, surely that's a breaking change, I'd vote to keep these, even if they act as a way to flag up changes in the underlying library when we come to upgrade.

As you say @robmarch2 documenting which parts of the library we are using would help someone if they came to swap it out too 👍

Reply:
@stephenwf I think the point @garyttierney is making is that as long as both the taxonomy terms and the input text get lemmatised according to the same scheme, the details of how that lemmatisation occurs (e.g. "I'm" -> "I be", "colors colours" -> "color colour", etc) shouldn't break any of our functionality. 

As far as I can tell, we'll only start getting issues if lemmatisation completely stops working (e.g. if we have "colour" in the taxonomy and "colours" in the input text, if lemmatisation stops working those terms won't match any more, whereas if the library were to change to lemmatise both to the American spelling they still will). So my view is that we need a test that fails in that case, but failures in the ones documenting the inner workings of the library won't help in reasoning about functional problems so we can drop them without losing anything

Reply:
This is a case where interacting with a mock is fine, and likely preferable. We don't care _what_ the result was, just that we got the result and we used it.

Reply:
I agree that using a mocked out pipeline would be fine for most of these tests that don't care about the output (e.g. `normaliseShouldTrimWhitespace`), but I want there to be tests that explicitly prove that the lemmatisation step of the pipeline occurs for English but not French since that's a fundamental part of what this class does. I've had a look into the `StanfordCoreNLP` class and I don't think we'll be able to prove that with mocks so I think we should definitely keep the English and French happy path tests as they currently are - I'm happy to change the other 4 over though if you feel strongly about it.